jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcInputFormat.java

```diff
@@ -66,7 +66,8 @@
         dbAccessor = DatabaseAccessorFactory.getAccessor(job);
       }
 
-      int numRecords = dbAccessor.getTotalNumberOfRecords(job);
+      //TODOY make sure I am not causing trouble 
+      int numRecords = numSplits <=1 ? Integer.MAX_VALUE : dbAccessor.getTotalNumberOfRecords(job);
```

Comment:
Could you explain this change?

Reply:
Yes, If we do not split than there is no need to query 'select count(*) from table' in order to decide how the limit the output query